test/interactions/dragInteractionTests.ts

```diff
@@ -1,376 +1,410 @@
+        assert.isFalse(endCallback1Called, "callback 1 was disconnected from the drag end interaction");
+        assert.isTrue(endCallback2Called, "callback 2 is still connected to the drag end interaction");
+
+        svg.remove();
+      });
+
+      it("can register multiple callbacks", () => {
+        let startCallbackCalled = false;
+        let moveCallbackCalled = false;
+        let endCallbackCalled = false;
+        let startCallback = () => startCallbackCalled = true;
+        let moveCallback = () => moveCallbackCalled = true;
+        let endCallback = () => endCallbackCalled = true;
+
+        dragInteraction.onDragStart(startCallback);
+        dragInteraction.onDrag(moveCallback);
+        dragInteraction.onDragEnd(endCallback);
+
+        dragInteraction.attachTo(component);
+
+        let target = component.background();
+        TestMethods.triggerFakeMouseEvent("mousedown", target, startPoint.x, startPoint.y);
+        assert.isTrue(startCallbackCalled, "callback for drag start was called");
+        TestMethods.triggerFakeMouseEvent("mousemove", target, endPoint.x, endPoint.y);
+        assert.isTrue(moveCallbackCalled, "callback for drag was called");
+        TestMethods.triggerFakeMouseEvent("mouseup", target, endPoint.x, endPoint.y);
+        assert.isTrue(endCallbackCalled, "callback for drag end was called");
+
+        startCallbackCalled = false;
+        moveCallbackCalled = false;
+        endCallbackCalled = false;
+        dragInteraction.offDragStart(startCallback);
+        dragInteraction.offDrag(moveCallback);
+
+        TestMethods.triggerFakeMouseEvent("mousedown", target, startPoint.x, startPoint.y);
+        assert.isFalse(startCallbackCalled, "callback for drag start was disconnected");
+        TestMethods.triggerFakeMouseEvent("mousemove", target, endPoint.x, endPoint.y);
+        assert.isFalse(moveCallbackCalled, "callback for drag was disconnected");
+        TestMethods.triggerFakeMouseEvent("mouseup", target, endPoint.x, endPoint.y);
+        assert.isTrue(endCallbackCalled, "callback for drag end is still connected");
+
+        svg.remove();
+      });
+
+      it("constraints the drag interaction to the Component space", () => {
+        let constrainedPos = { x: SVG_WIDTH, y: SVG_HEIGHT };
+        let constrainedNeg = { x: 0, y: 0 };
+
+        let receivedEnd: Plottable.Point;
+        dragInteraction.onDrag((startPoint, endPoint) => receivedEnd = endPoint);
+        dragInteraction.onDragEnd((startPoint, endPoint) => receivedEnd = endPoint);
+
+        dragInteraction.attachTo(component);
+        let target = component.content();
+
+        assert.isTrue(dragInteraction.constrainedToComponent(), "constrains by default");
+
+        TestMethods.triggerFakeMouseEvent("mousedown", target, startPoint.x, startPoint.y);
+        TestMethods.triggerFakeMouseEvent("mousemove", target, outsidePointPos.x, outsidePointPos.y);
```

Comment:
We could probably use `triggerFakeDragSequence` here, but it's clearer what's going on if we separate it out.